### PR TITLE
use CUDA_LINK_LIBRARIES_KEYWORD

### PR DESCRIPTION
**Pull request by headupinclouds**
* set(CUDA_LINK_LIBRARIES_KEYWORD PUBLIC)
* use target_link_libraries(… PUBLIC …)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -99,7 +99,8 @@ set(xgboost_exports "")
 set(xgboost_targets "")
 
 if(USE_CUDA)
-  
+
+  set(CUDA_LINK_LIBRARIES_KEYWORD PUBLIC)
```
Comment (ruslo) [inline CMakeLists.txt:103]: Not used?

Comment (headupinclouds) [reply CMakeLists.txt:103]: ... in `FindCUDA.cmake`
https://github.com/Kitware/CMake/blob/e40541339ae8a780bc410f0ef8b22d01e594c1ba/Modules/FindCUDA.cmake#L85-L90

else it uses the plain `target_link_libraries()` and we are forced to use that in the parent project for the same lib.

Comment (headupinclouds) [reply CMakeLists.txt:103]: And ... https://github.com/Kitware/CMake/blob/e40541339ae8a780bc410f0ef8b22d01e594c1ba/Modules/FindCUDA.cmake#L1822-L1827

Comment (ruslo) [reply CMakeLists.txt:103]: Okay, I see

**Pull request merged**